### ISSUE TITLE
Handle NA_STRING

### ISSUE THREAD

**Issue by qinwf** [qinwf/re2r#8]

**Comment by gagolews** [comment]
The desired strategy is to "return NA if the corresponding input is missing".

Btw, will the functions be vectorized with respect to both the "haystack" and the "needle" (e.g., n strings, 1 regex; 1 string, n regexes etc.?)

**Comment by qinwf** [comment]
Yes. Return the same result as `stringi` and `stringr` would be easier for users to learn and use. These two packages provide user-friendly APIs. 

I think I can learn from `stringi` package `stri__recycling_rule` function.

**Comment by gagolews** [comment]
sure, give it a try;

BTW, this is not an exported function but in some of the cases you may call stringi functions from C code directly, take a look at https://github.com/gagolews/ExampleRcppStringi

**Comment by tdhock** [comment]
about vectorizing "1 string, n regexes" -- what is the use case? I only use the other "n strings, 1 regex" version

**Comment by tdhock** [comment]
btw for the n string, n regexes case there is https://github.com/tdhock/revector
